code/datums/helper_datums/input.dm

```diff
@@ -0,0 +1,114 @@
+/proc/input_async(mob/user=usr, var/prompt, var/list/choices)
+	var/datum/async_input/A = new(choices, prompt)
+	A.show(user)
+	return A
+
+/proc/input_ranked_async(mob/user=usr, var/prompt="Order by greatest to least preference", var/list/choices)
+	var/datum/async_input/ranked/A = new(choices, prompt)
+	A.show(user)
+	return A
+
+/datum/async_input
+	var/datum/browser/popup
+	var/list/choices
+	var/flash = TRUE
+	var/immediate_submit = FALSE
+	var/prompt
+	var/result = null
+	var/style = "text-align: center;"
+	var/window_id
+	var/height = 200
+	var/width = 400
+
+/datum/async_input/New(var/list/choices, var/prompt="Pick an option:", var/window_id="async_input")
+	src.choices = choices
+	src.prompt = prompt
+	src.window_id = window_id
```

Comment:
please rename the arguments so that you don't need to resort to `src.`. `new_` should be a good prefix for this purpose (e.g. `new_choices`)